object[3] object_array
